processlib/services.py

```diff
@@ -99,6 +99,12 @@ def get_user_processes(user, include_unassigned=True):
             _activity_instances__assigned_group__isnull=True,
         ) & ~Q(_activity_instances__status=ActivityInstance.STATUS_CANCELED)
 
+    processes = Process.objects.filter(
+        status=Process.STATUS_STARTED,
+    ).filter(q).distinct()
+    for process in processes:
```

Comment:
I think we should rather construct a list of flows the user has access to and filter by those. If we run the permission checks on every object that could result in an enormous number of queries every time the list is accessed.

I guess the approach should be
1. use get_flows() to get all available flows
2. create a list of all flows where flow.permission is not set or the user has the flow.permission
3. filter the activity instances down so only those with the corresponding `process__flow_label` found above are listed

If we wanted to be precise we could also think of expanding this to look at the permissions of activity instances, too. But I think as a first step the above would  be fine.

Reply:
I have changed it to the list variant.